### PR TITLE
Fix expm1 gradient cancellation at large-negative inputs

### LINKED ISSUE #39794: jax.grad returns zero for expm1'(-40.0) instead of finite float64 derivative
### Description

`jax.grad` appears to return an incorrect zero gradient for `jax.numpy.expm1` at a large negative finite `float64` input.

The derivative of `expm1(x)` is `exp(x)`. For `x = -40.0`, the true derivative is approximately `4.248354255291589e-18`, which is a normal finite `float64` value. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math
import numpy as np

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = -40.0
expected = 4.248354255291589e-18

out = float(jax.grad(lambda z: jnp.expm1(z))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: 4.248354255291589e-18
```

### Expected result

```
4.248354255291589e-18
```

Since `d/dx expm1(x) = exp(x)`, the gradient at `x = -40.0` should be `exp(-40.0)`, approximately `4.248354255291589e-18`. This value is finite and representable in `float64`, so the backward pass should not flush it to zero.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

### LINKED ISSUE #39795: jax.nn.elu gradient returns zero for large negative float64 input
### Description

`jax.nn.elu` appears to return an incorrect zero gradient for a large negative finite `float64` input.

For the default ELU definition with `alpha = 1`, the negative branch is:

```
elu(x) = expm1(x), for x < 0
```

Therefore, for `x < 0`, the derivative should be:

```
elu'(x) = exp(x)
```

At `x = -40.0`, the true derivative is approximately `4.248354255291589e-18`, which is finite and representable in `float64`. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = -40.0
expected = 4.248354255291589e-18

out = float(jax.grad(lambda z: jax.nn.elu(z))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: 4.248354255291589e-18
```

### Expected result

```
4.248354255291589e-18
```

Since the negative branch of ELU is `expm1(x)`, its derivative at `x = -40.0` should be `exp(-40.0)`, approximately `4.248354255291589e-18`. This value is finite and representable in `float64`, so the backward pass should not flush it to zero.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

### LINKED ISSUE #39796: jax.nn.selu gradient returns zero for large negative float64 input
### Description

`jax.nn.selu` appears to return an incorrect zero gradient for a large negative finite `float64` input.

For the negative branch of SELU, the derivative is proportional to `exp(x)`. At `x = -700.0`, the true derivative is approximately `1.7334290832552396e-304`, which is finite and representable in `float64`. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = -700.0
expected = 1.7334290832552396e-304

out = float(jax.grad(lambda z: jax.nn.selu(z))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: 1.7334290832552396e-304
```

### Expected result

```
1.7334290832552396e-304
```

For the negative branch of SELU, the derivative should be a positive scaled `exp(x)` value. At `x = -700.0`, this value is still finite and representable in `float64`, so the backward pass should not flush it to zero.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

### PR DESCRIPTION
**Pull request by Phoebus-Liu**
## Description

The default (non-`HIGHEST`) gradient rule for `expm1_p` computed the derivative as `expm1(x) + 1`. For large-negative `x`, `expm1(x)` rounds to `-1.0`, so the sum cancels to exactly `0.0` and the gradient vanishes instead of equalling `exp(x)`.

This PR uses `exp(x)` directly as the analytic derivative for both accuracy modes, matching the existing `HIGHEST` branch.

### Example

```python
import jax, jax.numpy as jnp
jax.grad(lambda z: jnp.expm1(z))(jnp.asarray(-40.0))  # now ~4.25e-18, was 0.0
```

## Fixes

Fixes #39794
Fixes #39795
Fixes #39796

`elu` (#39795) and `selu` (#39796) compute their negative branch via `jnp.expm1`, so their gradients flow through this same `expm1_p` rule and are fixed by the same change.

## Test plan

- Added `LaxTest.testExpm1GradDefault` in `tests/lax_test.py`, which compares `grad(expm1)` against `grad(exp)` over a range in float32 (where the cancellation reproduces deterministically for `x ≲ -17`).
- Added `NNFunctionsTest.testEluGradLargeNegative` and `NNFunctionsTest.testSeluGradLargeNegative` in `tests/nn_test.py` (x64), checking the large-negative gradients equal `exp(x)` / `scale*alpha*exp(x)`.
- All of the above, plus the pre-existing `testExpm1Grad`, pass.

### CODE REVIEW

**Comment by hawkinsp** [comment]
This change looks reasonable, the main thing I'd want to do is see how disruptive it is likely to be to downstream users before landing it, which I can estimate by running our internal tests.

**Comment by hawkinsp** [comment]
While that runs, would you mind squashing your commits? Thanks!

**Comment by hawkinsp** [comment]
Yes, I think we can merge this. I just need the commits squashed!

**Comment by Phoebus-Liu** [comment]
Done — I've squashed the branch into a single commit; the net change is unchanged. Thanks for running the internal tests and for the review. This should now be ready to merge. Let me know if you'd like any tweaks to the commit message.

**Review by hawkinsp — APPROVED** [review]
